File: output/DebugServer.js
require('./mscorlib.js');require('./MongoDBLibrary.js');require('./CommonServerLibraries.js');require('./CommonLibraries.js');require('./CommonShuffleLibrary.js');require('./Models.js');
(function() {
	////////////////////////////////////////////////////////////////////////////////
	// DebugServer.DebugServer
	var $DebugServer_DebugServer = function() {
		CommonServerLibraries.Logger.start('Debug1');
		var fs = require('fs');
		var queueManager = new CommonShuffleLibrary.QueueManager('Debug1', new CommonShuffleLibrary.QueueManagerOptions([new CommonShuffleLibrary.QueueWatcher('DebugServer', null)], ['GatewayServer', 'Gateway*']));
		queueManager.addChannel('Area.Debug2.GetGameSource.Request', function(sender, data) {
			var sourceRequest = data;
			fs.readFile('/usr/local/src/new/Games/' + sourceRequest.gameName + '/app.js', 'ascii', function(err, data2) {
				queueManager.sendMessage(sender.gateway, 'Area.Debug.GetGameSource.Response', sender, { content: data2 });
			});
		});
	};
	$DebugServer_DebugServer.main = function() {
		try {
			new $DebugServer_DebugServer();
		}
		catch ($t1) {
			var exc = ss.Exception.wrap($t1);
			CommonServerLibraries.Logger.log('CRITICAL FAILURE: ' + CommonLibraries.ExtensionMethods.goodMessage(exc), 0);
		}
	};
	ss.registerClass(global, 'DebugServer.DebugServer', $DebugServer_DebugServer);
	$DebugServer_DebugServer.main();
})();

File: Servers/DebugServer/DebugServer.cs
using System;
using CommonLibraries;
using CommonServerLibraries;
using CommonShuffleLibrary;
using Models.GameManagerModels;
using NodeJSLibrary;
namespace DebugServer
{
    public class DebugServer
    {
        public DebugServer()
        {
            Logger.Start("Debug1");

            var fs = Global.Require<FS>("fs");

            var queueManager = new QueueManager("Debug1",
                                                new QueueManagerOptions(new[] {
                                                                                      new QueueWatcher("DebugServer", null),
                                                                              },
                                                                        new[] {"GatewayServer", "Gateway*"}));

            queueManager.AddChannel("Area.Debug2.GetGameSource.Request",
                                    (sender, data) => {
                                        var sourceRequest = (GameSourceRequestModel) data;
                                        fs.ReadFile(ExtensionMethods.HARDLOCATION+"Games/" + sourceRequest.GameName + "/app.js",
                                                    "ascii",
                                                    (err, data2) => { queueManager.SendMessage(sender.Gateway, "Area.Debug.GetGameSource.Response", sender, new GameSourceResponseModel(data2)); });
                                    });
        }

        public static void Main()
        {
            try {
                new DebugServer();
            } catch (Exception exc) {
                Logger.Log("CRITICAL FAILURE: " + exc.GoodMessage(), LogLevel.Error);
            }
        }
    }
}